contrib/epee/include/net/abstract_http_client.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include <boost/optional/optional.hpp>
+#include "http_base.h"
+#include "http_auth.h"
+#include "net/net_ssl.h"
+#include "net_parse_helpers.h"
+
+#undef MONERO_DEFAULT_LOG_CATEGORY
+#define MONERO_DEFAULT_LOG_CATEGORY "net.http"
+
+namespace epee
+{
+  namespace net_utils
+  {
+    //---------------------------------------------------------------------------
+    static inline const char* get_hex_vals()
+    {
+      static const char hexVals[16] = {'0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F'};
+      return hexVals;
+    }
+
+    static inline const char* get_unsave_chars()
+    {
+      //static char unsave_chars[] = "\"<>%\\^[]`+$,@:;/!#?=&";
+      static const char unsave_chars[] = "\"<>%\\^[]`+$,@:;!#&";
+      return unsave_chars;
+    }
+
+    static inline bool is_unsafe(unsigned char compare_char)
```

Comment:
Take the opportunity to put every function from here down into a cpp. Drop `static inline` from every function signature.